server_windows.go

```diff
@@ -37,3 +45,80 @@ func (s *Server) toLocalPath(p string) string {
 
 	return lp
 }
+
+func bitsToDrives(bitmap uint32) []string {
+	var drive rune = 'a'
+	var drives []string
+
+	for bitmap != 0 {
+		if bitmap&1 == 1 {
+			drives = append(drives, string(drive)+":")
+		}
+		drive++
+		bitmap >>= 1
+	}
+
+	return drives
+}
+
+func getDrives() ([]string, error) {
+	mask, err := windows.GetLogicalDrives()
+	if err != nil {
+		return nil, fmt.Errorf("GetLogicalDrives: %w", err)
+	}
+	return bitsToDrives(mask), nil
+}
+
+type driveInfo struct {
+	fs.FileInfo
+	name string
+}
+
+func (i *driveInfo) Name() string {
+	return i.name // since the Name() returned from a os.Stat("C:\\") is "\\"
+}
+
+type winRoot struct {
+	dummyFile
+	doneDirs int
```

Comment:
We could also just store a `drives []string` that is filled in at `openFile` and then pop off the return values as they are read:

```
drives = f.drives
if len(drives) > n {
  drives = drives[:n]
}
f.drives = f.drives[:len(drives)]
```

Then we don’t have to make a syscall every time, and potentially receiving different results because say, a new drive attached or not.

Reply:
I think it should be `f.drives = f.drives[:len(drives)]`?

Reply:
fixed!